templates/sfn/ingest_sfn_definition.json.tpl

```diff
@@ -495,7 +495,42 @@
           }
         }
       },
+      "Next": "Get number of items in lock table that have this batchId"
+    },
+    "Get number of items in lock table that have this batchId": {
+      "Type": "Task",
+      "Parameters": {
+        "TableName": "${ingest_lock_table_name}",
+        "IndexName": "${ingest_lock_table_gsi_name}",
+        "KeyConditionExpression": "batchId = :lookUpId",
+        "ExpressionAttributeValues": {
+          ":lookUpId": {
+            "S.$": "$$.Execution.Input.batchId"
+          }
+        }
+      },
+      "Resource": "arn:aws:states:::aws-sdk:dynamodb:query",
+      "Next": "Check if number of items is 0"
+    },
+    "Check if number of items is 0": {
+      "Type": "Choice",
+      "Choices": [
+        {
+          "Variable": "$.ScannedCount",
```

Comment:
Although `Count` and `ScannedCount` [will be the same](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_Query.html#API_Query_ResponseElements:~:text=If%20you%20did%20not%20use%20a%20filter%20in%20the%20request%2C%20then%20ScannedCount%20is%20the%20same%20as%20Count.) as we're not applying a filter, can we use `Count` here to future-proof a filter being added.